### PR TITLE
EVG-8963: implement test results gRPC client

### PR DESCRIPTION
**Pull request by Kimchelly**
JIRA: https://jira.mongodb.org/browse/EVG-8963

Also some minor linter fixes.

### CODE REVIEW

File: testresults/client.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+	Trial       int32
+	Status      string
+	LogURL      string
+	LineNum     int32
+	TaskCreated time.Time
+	TestStarted time.Time
+	TestEnded   time.Time
+}
+
+// Export converts a Result into the equivalent protobuf TestResult.
```
Comment (julianedwards) [inline testresults/client.go:181]: `Export` -> `export`

File: testutil/server.go [hunk 2]
```diff
@@ -29,8 +31,65 @@ type MockMetricsServer struct {
...
+	}()
+	go func() {
+		<-ctx.Done()
+		s.Stop()
+	}()
+	return srv, nil
+}
+
+// NewMockMetricsServerWithDialOpts will return a new MockMetricsServer listening
+// on the port and url from the specified dial options
```
Comment (julianedwards) [inline testutil/server.go:64]: missing period

**Review by julianedwards — CHANGES_REQUESTED** [review]
Can we move the types and export functions out of `client.go` and into a separate file?

File: testresults/client.go [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+	}
+
+	s := &Client{
+		client:    internal.NewCedarTestResultsClient(conn),
+		closeConn: func() error { return nil },
+	}
+	return s, nil
+}
+
+type CreateOptions struct {
```
[the remark this answers is not in the document]
Comment (Kimchelly) [reply testresults/client.go]: Done.

File: testresults/client.go [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+	Project     string `bson:"project" json:"project" yaml:"project"`
+	Version     string `bson:"version" json:"version" yaml:"version"`
+	Variant     string `bson:"variant" json:"variant" yaml:"variant"`
+	TaskID      string `bson:"task_id" json:"task_id" yaml:"task_id"`
+	TaskName    string `bson:"task_name" json:"task_name" yaml:"task_name"`
+	Execution   int32  `bson:"execution" json:"execution" yaml:"execution"`
+	RequestType string `bson:"request_type" json:"request_type" yaml:"request_type"`
+	Mainline    bool   `bson:"mainline" json:"mainline" yaml:"mainline"`
+}
```
[the remark this answers is not in the document]
Comment (Kimchelly) [reply testresults/client.go]: I don't think so, but I copied these fields from system metrics, which seems to have copied it from buildlogger. Though, I don't really see why buildlogger needs them either. I removed them from system metrics and test results, but not buildlogger.

File: testresults/client.go (hunk 1 above)
Comment (Kimchelly) [reply testresults/client.go]: Done.

File: testutil/server.go (hunk 2 above)
Comment (Kimchelly) [reply testutil/server.go:64]: Fixed.

**Review by Kimchelly — COMMENTED** [review]
I moved the types to separate files and did the same thing for system metrics.

**Review by julianedwards — APPROVED** [review]

**Pull request merged**